lib/puppet/provider/package/pip.rb

```diff
@@ -56,7 +56,13 @@ def query
   # cache of PyPI's package list so this operation will always have to
   # ask the web service.
   def latest
-    client = XMLRPC::Client.new2("http://pypi.python.org/pypi")
+    proxy = ENV["http_proxy"]
+    if proxy.nil? then
+      client = XMLRPC::Client.new2("http://pypi.python.org/pypi")
+    else
+      host, port = proxy[7..-1].split(':')
+      client = XMLRPC::Client.new2("http://pypi.python.org/pypi", proxy_host=host, proxy_port=port)
```

Comment:
Right, but the names here have no bearing on the arguments:

```ruby
def foo(a, b)
  puts a, b
end

foo(c = 1, d = 2)
puts c
puts d
```

Results in:
```
1
2
1
2
```

The above is simply assigning `c` to `1` and `d` to `2` and passing the result of the expressions (the current values of `c` and `d` respectively) to the function.